This notebook verifies if the various simulated quantities are realistic for observations. 

Converting subhalo (galaxy) luminosity to apparent magnitude

In [1]:
%autoreload 2 
%matplotlib inline

In [2]:
import sys
sys.path.append("../")

In [3]:
from calculate_astrophy_quantities import *

# get illustris cosmology 

In [4]:
illustris_cosmology = get_Illustris_cosmology()

# How bright Illustris galaxies need to be for them to be observable

Keck has a limiting magnitude of 24.4 

In [5]:
z_range = np.arange(0.5, 3., 0.1)
limiting_apparent_magnitude = \
    map(lambda z: dimmest_mag_for_illustris_halos_to_be_observable(
        illustris_cosmology, z=z), z_range)

In [6]:
limiting_apparent_magnitude = np.array(limiting_apparent_magnitude)

In [7]:
limiting_apparent_magnitude

array([-18.26963556, -18.75097312, -19.16214971, -19.52084283,
       -19.83869917, -20.12383743, -20.38216278, -20.61811465,
       -20.83511967, -21.03588021, -21.22256585, -21.396945  ,
       -21.56047785, -21.71438382, -21.85969138, -21.99727565,
       -22.12788718, -22.25217426, -22.37070052, -22.48395884,
       -22.59238265, -22.69635494, -22.79621573, -22.89226821, -22.98478376])

In [16]:
plt.plot(z_range, limiting_apparent_magnitude)

NameError: name 'plt' is not defined

In [37]:
24.4 - 5. * (np.log10(lum_distance.value * 1e6) - 1.)

-16.555400636223652

This means that the galaxies have to have an apparent magnitude of at most -16.6
$$ m <= -16.6 $$
to be observable

# Field of view (FOV) of different telescopes

assuming z = 0.3

In [27]:
view = {"width": 34., "height": 27}

In [38]:
subaru_field_of_view(Illustris_cosmology, view=view, z=0.3)

array([ 9.08949905,  7.2181316 ])

Most clusters seem to fit comfortably within the FOV of Subaru

In [39]:
view = {"width": 3., "height": 3.}

In [40]:
Hubble_acs_field_of_view(Illustris_cosmology, view=view, z=0.3)

array([ 0.80201462,  0.80201462])

Observers may need to have multiple pointings if they want to do weak / strong lensing with the HST